File: python/triangle.py
def minimumTotal(self, triangle):
        """
        :type triangle: List[List[int]]
        :rtype: int
        """
        if len(triangle) == 0:
            return -1
        if len(triangle) == 1:
            return triangle[0][0]
        
        l = len(triangle)-1
        
        for i in range(l)[::-1]:
            for j in range(i+1):
                triangle[i][j] += min(triangle[i+1][j],triangle[i+1][j+1])
        
        return triangle[0][0]

File: C_plus/Triangle.cpp
int minimumTotal(vector<vector<int>>& triangle) {
        
        
        if (triangle.empty()) return -1;
        
        
        int l1 = triangle.size()-1;
        
        for(int i=l1-1;i>=0;i--) {
            
            for(int j = 0; j<=i;j++) {
                
                triangle[i][j] += min(triangle[i+1][j], triangle[i+1][j+1]); 
                
                
            }
            
        }
        
        return triangle[0][0];
        
        
    }
